src/Qir/check-sources-formatted.ps1

```diff
@@ -0,0 +1,47 @@
+#Requires -PSEdition Core
+
+param (
+    [Parameter()]
+    [String]
+    $DirPath = "$PSScriptRoot"
+)
+
+$tmpFile = "format.log"
+
+#Write-Host "1"
+#"$DirPath/*.cpp","$DirPath/*.c","$DirPath/*.h","$DirPath/*.hpp" | get-childitem -Recurse `
+#    | ?{$_.fullname -notlike "*\Externals\*"} | ?{$_.fullname -notlike "*\drops\*"} | ?{$_.fullname -notlike "*\bin\*"} `
+#    | %{clang-format -n -style=file $_.fullname}
+#
+#Write-Host "2"
+#"$DirPath/*.cpp","$DirPath/*.c","$DirPath/*.h","$DirPath/*.hpp" | get-childitem -Recurse `
+#    | ?{$_.fullname -notlike "*\Externals\*"} | ?{$_.fullname -notlike "*\drops\*"} | ?{$_.fullname -notlike "*\bin\*"} `
+#    | %{clang-format -n -style=file $_.fullname} 2>format.log
+
+#Write-Host "3"
+#$filesRequireFormatting = get-content $tmpFile | ?{$_ -like "*: warning:*"} `
+#                            | %{[string]::join(":",($_.split("warning:")[0].split(":") | select -SkipLast 3))} `
+#                            | sort | unique
+#Remove-Item $tmpFile
+
+Write-Host "A"
+&{
+   Write-Warning "warning"
+   Write-Error "error"
```

Comment:
@kuzminrobin: If I recall, the build pipeline is currently configured to interpret any text written to stderr as a failure (similar to `-o pipefail` in bash). The trick to avoid that is to turn stderr into stdout, but then that can also be interpreted as an error unless `$ErrorPreference` is set to `"SilentlyContinue"` (https://stackoverflow.com/a/66726535). As detailed in that answer, you can also use `#Requires` to depend on 7.1, and use the `PSNotApplyErrorActionToStderr` experimental feature to turn off interpreting stderr as an exception stream.